app/src/main/java/de/tum/in/tumcampusapp/api/tumonline/TUMOnlineClient.kt

```diff
@@ -0,0 +1,47 @@
+package de.tum.`in`.tumcampusapp.api.tumonline
+
+import android.content.Context
+import com.tickaroo.tikxml.TikXml
+import com.tickaroo.tikxml.retrofit.TikXmlConverterFactory
+import de.tum.`in`.tumcampusapp.api.app.Helper
+import retrofit2.Retrofit
+
+class TUMOnlineClient {
+
+    companion object {
+
+        private const val BASE_URL = "https://campus.tum.de/tumonline/"
+
+        private var apiService: TUMOnlineAPIService? = null
+
+        @JvmStatic fun getInstance(context: Context): TUMOnlineAPIService {
+            if (apiService == null) {
```

Comment:
I think this method needs to be synchronized to avoid race conditions for two concurrent calls to `getInstance()`